web/src/hooks/useCookie.ts

```diff
@@ -0,0 +1,33 @@
+import { DateTime } from 'luxon'
+import { useState, useCallback } from 'react'
+
+import Cookie from '../utils/Cookie'
+
+type UseCookieReturnType<T> = {
+  value: T
+  updateCookie: (newValue: T, expires?: DateTime, domain?: string, path?: string) => void
+  deleteCookie: (path?: string, domain?: string) => void
+}
+const useCookie = <T>(name: string): UseCookieReturnType<T> => {
+  const [value, setValue] = useState<string | null>(Cookie.get(name))
```

Comment:
Optional: You could also save the value directly to avoid a lot of JSON.parse/JSON.stringify:
```suggestion
  const [value, setValue] = useState<T | null>(JSON.parse(Cookie.get(name) ?? 'null'))
```
Warning: See some other comments, this would break rerendering if you are continuing mutation instead of spread syntax :)